rct229/rules/section5/section5rule37.py

```diff
@@ -0,0 +1,258 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.data_fns.table_G3_4_fns import table_G34_lookup
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_building_scc_skylight_roof_ratios_dict import (
+    get_building_scc_skylight_roof_ratios_dict,
+)
+from rct229.ruleset_functions.get_opaque_surface_type import OpaqueSurfaceType as OST
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.std_comparisons import std_equal
+
+MANUAL_CHECK_MSG = "Manual review is required to verify skylight meets U-factor requirement as per table G3.4."
+MANUAL_CHECK_APPLICABLE = (
+    "The subsurface type is Door, not applicable for the rule-checking"
+)
+DOOR = schema_enums["SubsurfaceClassificationType"].DOOR.name
+
+
+class Section5Rule37(RuleDefinitionListIndexedBase):
+    """Rule 37 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule37, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule37.BuildingRule(),
+            index_rmr="baseline",
+            id="5-37",
+            description="Skylight U-factors for residential, non-residential and semi-heated spaces in the baseline model must match the appropriate requirements in Table G3.4-1 through G3.4-8.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {
+            "climate_zone": rmr_baseline["weather"]["climate_zone"],
+        }
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule37.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section5Rule37.BuildingRule.RoofRule(),
+                index_rmr="baseline",
+                list_path="$..surfaces[*]",
+                manual_check_required_msg=MANUAL_CHECK_MSG,
+            )
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
```

Comment:
I would prefer that we order the methods in the order in which they are called. In this case it would be

- create_data
- manual_check_required
- list_filter

Reply:
Isn't it manual_check_required go first?
In evaluate function of a RuleDefinitionListIndexedBase class, `manual_check_required` is called first, and then calls the `rule_check`, inside `rule_check`, it calls `create_data` first and then `create_context_list`.
Right after the `create_context_list`, we called `list_filter` to go into the next level `evaluate` function.

So I think the order should be
- manual_check_required
- create_data
- list_filter
?